proto/pulumi/analyzer.proto

```diff
@@ -43,6 +43,45 @@ service Analyzer {
     rpc GetPluginInfo(google.protobuf.Empty) returns (PluginInfo) {}
     // Configure configures the analyzer, passing configuration properties for each policy.
     rpc Configure(ConfigureAnalyzerRequest) returns (google.protobuf.Empty) {}
+
+    // `Handshake` is the first call made by the engine to an analyzer. It is used to pass the engine's address to the
+    // analyzer so that it may establish its own connections back, and to establish protocol configuration that will be
+    // used to communicate between the two parties.
+    rpc Handshake(AnalyzerHandshakeRequest) returns (AnalyzerHandshakeResponse) {}
+}
+
+
+// `AnalyzerHandshakeRequest` is the type of requests sent as part of a [](pulumirpc.Analyzer.Handshake) call.
+message AnalyzerHandshakeRequest {
+    // The gRPC address of the engine handshaking with the analyzer. At a minimum, this address will expose an instance
+    // of the [](pulumirpc.Engine) service.
+    string engine_address = 1;
+
+    // A *root directory* where the analyzer's binary, `PulumiPolicy.yaml`, or other identifying source code is located.
+    // In the event that the analyzer is *not* being booted by the engine (e.g. in the case that the engine has been
+    // asked to attach to an existing running analyzer instance via a host/port number), this field will be empty.
+    optional string root_directory = 2;
+
+    // A *program directory* in which the analyzer should execute. This is generally a subdirectory of the root
+    // directory, though this is not required. In the event that the analyzer is *not* being booted by the engine (e.g.
+    // in the case that the engine has been asked to attach to an existing running analyzer instance via a host/port
+    // number), this field will be empty.
+    optional string program_directory = 3;
```

Comment:
> This is generally a subdirectory of the root directory

I was thinking this would be the directory of the program we're analysing. Is the plan to run the analyser with `CWD` set to the program? In that case I find the message `A *program directory* in which the analyzer should execute.` slightly confusing.

Reply:
> Is the plan to run the analyser with CWD set to the program?

Yes for the new plugins via RunPlugin, that matches what we do for provider plugins.

> slightly confusing.

Yeh this is the same wording we have for provider handshake 😕 I'm not sure how to well distinguish between "pulumi program" and "plugin program" when they're both programs.